lib/shared-method.js

```diff
@@ -608,6 +614,29 @@ SharedMethod.prototype.isDelegateFor = function(suspect, isStatic) {
 };
 
 /**
+ * Determine if this shared method invokes the given "suspect" function.
+ *
+ * @example
+ * ```js
+ * sharedMethod.isDelegateForName(myClass.myMethod); // pass a function
+ * sharedMethod.isDelegateForName(myClass.prototype.myInstMethod);
+ * sharedMethod.isDelegateForName('myMethod'); // check for a static method by name
+ * sharedMethod.isDelegateForName('prototype.myInstMethod'); // instance method by name
+ * ```
+ *
+ * @returns Boolean True if the shared method invokes the given function; false otherwise.
+ */
+
+SharedMethod.prototype.isDelegateForName = function(suspect) {
+  var isStatic = !/^prototype\./.test(suspect);
+  var baseName = suspect;
+  if (!isStatic) {
+    baseName = suspect.substring(10);
+  }
```

Comment:
This is sort of repeating, as `10` is tied to the value in the regexp on the first line.

I think this is more elegant:

```js
var m = suspect.match(/^prototype\.(.*)$/);
var isStatic = !m;
var baseName = isStatic ? suspect : m[1];
```